nifi-api/src/main/java/org/apache/nifi/processor/AbstractProcessor.java

```diff
@@ -27,7 +30,12 @@ public final void onTrigger(final ProcessContext context, final ProcessSessionFa
             onTrigger(context, session);
             session.commit();
         } catch (final Throwable t) {
-            getLogger().error("{} failed to process due to {}; rolling back session", new Object[]{this, t});
+            StringWriter stacktraceWriter = new StringWriter();
```

Comment:
This logic needs to be in the logger and that is its intent.  You will note that the throwable is being passed into it.  The stack traces are getting logged when debug level is engaged for a given logger.  This allows the user to control whether they are generated or not.  That said...it is clearly counterintuitive and we need to improve.